packages/driver/src/cy/logGroup.ts

```diff
@@ -0,0 +1,57 @@
+import { $Command } from '../cypress/command'
+import $errUtils from '../cypress/error_utils'
+import type { LogConfig } from '../cypress/log'
+
+export interface LogGroupConfig {
+  // name of the group - defaults to current command's name
+  name?: string
+  // additional information to include in the log
+  message?: any
+  // the JQuery element for the command. This will highlight the command
+  // in the main window when debugging
+  $el?: JQuery
+  // whether or not to emit a log to the UI
+  // when disabled, child logs will not be nested in the UI
+  log?: boolean
+  // timeout of the group command - defaults to defaultCommandTimeout
+  timeout?: number
+}
+
+export default (Cypress, userOptions: LogGroupConfig, fn: (log) => any) => {
+  const cy = Cypress.cy
+
+  const shouldEmitLog = userOptions.log || true
+
+  const options: LogConfig = {
+    ...userOptions,
+    instrument: 'command',
+    groupStart: true,
+    emitOnly: !shouldEmitLog,
+  }
+
+  const subject = userOptions.$el || null
+
+  const log = Cypress.log(options)
```

Comment:
When `Cypress.log` has [`emitOnly: true`](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/log.ts#L601) it will not send emit the log to the runner/reporter and will not return a log instance. It will however, create the log attributes and store it in the LogManager and add the logs on the Command. This seems to work just fine, however,

- Is there risk of data leaking when this test/command is sent to the dashbaord? I didn't find this attribute in the [schema](https://github.com/cypress-io/json-schemas/blob/master/src/schemas/test-runner-test.ts), but wanted to double check.
- Is there risk of this behaving unexpectedly if the user were to pass `{ log: false }` to the command consuming this?

For example:
```
cy.get('#form').within({ log: false}, () => {})
```
This example is likely less applicable to `.within()` and more applicable to up-coming consumption in `.withToDomain()`.